karamel-core/src/main/java/se/kth/karamel/backend/github/util/ChefExperimentExtractor.java

```diff
@@ -0,0 +1,296 @@
+        recipeDepsKaramelfile.append(lDepsFinal.toString());
+        recipeDepsKaramelfile.append("    global:").append(System.lineSeparator());
+        recipeDepsKaramelfile.append(gDepsFinal.toString());
+      }
+
+      StringBuilder karamelContents = CookbookGenerator.instantiateFromTemplate(
+          Settings.CB_TEMPLATE_KARAMELFILE,
+          "name", repoName,
+          "next_recipes", recipeDepsKaramelfile.toString()
+      );
+      // TODO - integration with cluster defn file
+//      String ymlString = experimentContext.getClusterDefinition();
+
+      String berksfile = experimentContext.getBerksfile();
+
+      StringBuilder berksContents = CookbookGenerator.instantiateFromTemplate(
+          Settings.CB_TEMPLATE_BERKSFILE,
+          "berks_dependencies", berksfile
+      );
+
+      Github.addFile(owner, repoName, "Berksfile", berksContents.toString());
+
+      Map<String, String> expConfigFileNames = new HashMap<>();
+      Map<String, String> expConfigFilePaths = new HashMap<>();
+
+      // 2. write them to recipes/default.rb and metadata.rb
+      for (Code experiment : experiments) {
+        String experimentName = experiment.getName();
+        String configFilePath = experiment.getConfigFileName();
+        String configFileContents = experiment.getConfigFileContents();
+
+        String configFileName = configFilePath;
+        int filePos = configFileName.lastIndexOf("/");
+        if (filePos != -1) {
+          configFileName = configFileName.substring(filePos + 1);
+        }
+
+        String email = (Github.getEmail() == null) ? "karamel@karamel.io" : Github.getEmail();
+
+        StringBuilder recipe_rb = CookbookGenerator.instantiateFromTemplate(
+            Settings.CB_TEMPLATE_RECIPE_EXPERIMENT,
+            "cookbook", repoName,
+            "name", experimentName,
+            "interpreter", experiment.getScriptType(),
+            "user", experimentContext.getUser(),
+            "group", experimentContext.getGroup(),
+            "script_contents", experiment.getScriptContents()
+        );
+
+        String recipeContents = recipe_rb.toString();
+
+        // Replace all parameters with chef attribute values
+        for (String attr : attrs.keySet()) {
+          recipeContents = recipeContents.replaceAll("%%" + attr + "%%", "#{node[:" + repoName + "][:" + attr + "]}");
+        }
+
+        for (String attr : attrs.keySet()) {
+          configFileContents = configFileContents.replaceAll("%%" + attr + "%%",
+              "<%= node[:" + repoName + "][:" + attr + "] =>");
+        }
+
+        if (!configFilePath.isEmpty()) {
+          expConfigFileNames.put(experimentName, configFileName);
+          expConfigFilePaths.put(experimentName, configFilePath);
+        }
+
+        // 3. write them to files and push to github
+        Github.addFile(owner, repoName, "recipes" + File.separator + experimentName + ".rb", recipeContents);
```

Comment:
No. This is  a file on the local filesystem. The CookbookScaffolder class creates a chef cookbook with files. We then need to add them and commit and push them to github. On windows, you should use the file separator. But, yes we should be careful.
Slash = URLs
File.separator = Local files (cloned repo)